ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/messages/BlockBodiesMessageTest.java

```diff
@@ -65,16 +81,37 @@ public void blockBodiesRoundTrip() throws IOException {
     final MessageData initialMessage = BlockBodiesMessage.create(bodies);
     final MessageData raw = new RawMessage(EthPV62.BLOCK_BODIES, initialMessage.getData());
     final BlockBodiesMessage message = BlockBodiesMessage.readFrom(raw);
-    final Iterator<BlockBody> readBodies =
-        message
-            .bodies(
-                FixedDifficultyProtocolSchedule.create(
-                    GenesisConfigFile.development().getConfigOptions(),
-                    false,
-                    EvmConfiguration.DEFAULT))
-            .iterator();
+    final Iterator<BlockBody> readBodies = message.bodies(protocolSchedule).iterator();
     for (int i = 0; i < 50; ++i) {
       Assertions.assertThat(readBodies.next()).isEqualTo(bodies.get(i));
     }
   }
+
+  @Test
+  public void shouldEncodeEmptyBlocksInBlockBodiesMessage() {
+    final Bytes bytes = Bytes.fromHexString("0xc2c0c0");
+    final MessageData raw = new RawMessage(EthPV62.BLOCK_BODIES, bytes);
+    final BlockBodiesMessage message = BlockBodiesMessage.readFrom(raw);
+    final List<BlockBody> bodies = message.bodies(protocolSchedule);
+    bodies.forEach(blockBody -> Assertions.assertThat(blockBody.isEmpty()).isTrue());
+  }
+
+  @Test
+  public void shouldNotThrowRLPExceptionIfAllowedEmptyBody() {
+    final Bytes bytes = Bytes.fromHexString("0xc0");
+    final BlockBody empty = BlockBody.readFrom(RLP.input(bytes), null, true);
+    Assertions.assertThat(empty.isEmpty()).isTrue();
+  }
+
+  @Test
+  public void shouldThrowRLPExceptionIfNotAllowedEmptyBody() {
+    final Bytes bytes = Bytes.fromHexString("0xc0");
+    final BlockHeaderFunctions blockHeaderFunctions =
+        ScheduleBasedBlockHeaderFunctions.create(protocolSchedule);
```

Comment:
is the blockHeaderFunctions used?

Reply:
it's not created for the other test shouldNotThrowRLPExceptionIfAllowedEmptyBody